### PR TITLE
Add "Ratio"

### PR DESCRIPTION
**Pull request by WaffleLapkin**
This PR adds ratio (the difference between `km`&`m` or `min`&`h`) into the lib.

### CODE REVIEW

File: src/eq.rs [hunk 1]
```diff
@@ -28,5 +30,25 @@ where
...
+mod sealed {
+    use crate::fraction::Fraction;
+    use core::ops::Mul;
+
+    pub trait FractionEq<Rhs> {}
+
+    impl<A, B, U, V> FractionEq<Fraction<A, B>> for Fraction<U, V>
+    where
+        A: Mul<V>,
+        U: Mul<B, Output = A::Output>,
```
Comment (Mr-Andersen) [inline src/eq.rs:52]: Damn, I took some time to understand these two lines :sweat_smile:

File: src/fraction.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    /// ```
+    /// use typed_phy::{fraction::FractionTrait, Frac};
+    /// use typenum::{U5, U7};
+    ///
+    /// assert_eq!(<Frac![U5 / U7]>::mul(14), 10)
+    /// ```
+    #[inline]
+    fn mul<I>(int: I) -> I
+    where
+        I: FromUnsigned + Mul<Output = I> + Div<Output = I>,
```
Comment (Mr-Andersen) [inline src/fraction.rs:54]: Probably you want to implement multiplication/division resulting in `f64`/`f32`?

File: src/prefixes.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+/// Multiplies ratio of `U` by `X`
+pub(crate) type MulPow10<U, P> = MulBy<U, <U10 as Pow<P>>::Output>;
+/// Divides ratio of `U` by `X`
+pub(crate) type DivPow10<U, P> = DivBy<U, <U10 as Pow<P>>::Output>;
+
+/// yotta prefix. `Y`. (Base 10: `10^24`, decimal: `1000000000000000000000000`,
+/// word: septillion/quadrillion, adoption: 1991)
+pub type Yotta<U> = MulPow10<U, U24>;
+/// zetta prefix. `Z`. (Base 10: `10^21`, decimal: `1000000000000000000000`,
+/// word: sextillion/trilliard, adoption: 1991)
```
Comment (Mr-Andersen) [inline src/prefixes.rs:16]: Omg adoption year

File: src/simplify.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+/// ## Examples
+/// ```
+/// use typed_phy::{fraction::Fraction, simplify::Simplify};
+///
+/// use typenum::{assert_type_eq, U12, U3, U32, U8};
+///
+/// type Simplified = <Fraction<U32, U12> as Simplify>::Output;
+/// assert_type_eq!(Simplified, Fraction::<U8, U3>);
+/// ```
+pub trait Simplify {
```
Comment (Mr-Andersen) [inline src/simplify.rs:16]: Btw I didn't check GCD because it's big :neutral_face:

File: src/unit.rs [hunk 5]
```diff
@@ -144,7 +150,7 @@ where
 /// `Unit<1, 0, -1, ...> - Unit<0, 0, 1, ...> = Unit<1, 0, -2, ...>`
```
Comment (Mr-Andersen) [inline src/unit.rs]: Replace `-` with `/` (same with `+ => *` upper)

**Review by Mr-Andersen — APPROVED** [review]
Nice :star:
(There are some minor things tho)

File: src/eq.rs (hunk 1 above)
Comment (WaffleLapkin) [reply src/eq.rs:52]: :sweat_smile: this probably needs an explanation comment

Comment (Mr-Andersen) [reply src/eq.rs:52]: I dunno. It's probably obvious for experienced people that `A::Output` is deduced to `<A as Mul<V>>::Output`, it just wasn't obvious for me

File: src/fraction.rs (hunk 2 above)
Comment (WaffleLapkin) [reply src/fraction.rs:54]: Sure. I forgot about `FromUnsigned` impls for `f64`/`f32`.

File: src/prefixes.rs (hunk 3 above)
Comment (WaffleLapkin) [reply src/prefixes.rs:16]: just copypasted from Wikipedia xd

File: src/simplify.rs (hunk 4 above)
Comment (WaffleLapkin) [reply src/simplify.rs:16]: Ok. GCD is a very hard feature :sweat_smile:

File: src/unit.rs (hunk 5 above)
Comment (WaffleLapkin) [reply src/unit.rs]: Oh, nice catch!

File: src/prefixes.rs (hunk 3 above)
Comment (Mr-Andersen) [reply src/prefixes.rs:16]: Thank god

**Review by Mr-Andersen — APPROVED** [review]

**Pull request merged**